rustworkx-core/src/generators/binomial_tree_graph.rs

```diff
@@ -132,30 +115,26 @@ where
             let source_index = source + n;
             let target_index = target + n;
 
-            if !find_edge(&mut graph, source_index, target_index) {
-                graph.add_edge(
-                    graph.from_index(source_index),
-                    graph.from_index(target_index),
-                    default_edge_weight(),
-                );
-            }
-            if bidirectional && !find_edge(&mut graph, target_index, source_index) {
-                graph.add_edge(
+            graph.update_edge(
+                graph.from_index(source_index),
+                graph.from_index(target_index),
+                default_edge_weight(),
+            );
```

Comment:
This isn't quite the same because if the edge already exists we're replacing the weight with the output of `default_edge_weight()`. This won't matter if `default_edge_weight()` returns the same value every time, but there is no guarantee that will be the case.

Reply:
Ah, missed that. Was thinking of the defaults as fixed. I can return to doing the find and then the add if not there.

Reply:
Done in https://github.com/Qiskit/rustworkx/pull/784/commits/63d2307d1fd505b22ef802a407d05b26b27e08a9.